src/components/symptom-shark/presentational/LogEntryIcon/LogEntryIcon.tsx

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import { FontAwesomeIcon, FontAwesomeIconProps } from "@fortawesome/react-fontawesome";
+import {  faStar } from "@fortawesome/free-regular-svg-icons";
+import { faAmbulance, faBolt, faFlask, faSpoon, faStar as faStarSolid, faThermometer, faTrophy, faFlag, faHospital, faMoon, faQuestionCircle } from "@fortawesome/free-solid-svg-icons";
+import { faUserMd } from "@fortawesome/free-solid-svg-icons";
+
+export type LogEntryIconKey = "star" | "star-o" | "user-md" | "hospital-o" | "ambulance" | "flag" | "moon-o" | "flask" | "bolt" | "question-circle-o" | "thermometer" | "trophy" | "spoon";
+
+export interface LogEntryIconProps {
+    icon?: LogEntryIconKey;
+    className?: string;
+}
+
+export default function LogEntryIcon(props: LogEntryIconProps) {
+    let iconProps = { ...logEntryIcons[props.icon || "star-o"], className: props.className };
+    return <FontAwesomeIcon {...iconProps} />;
+}
+
+const logEntryIcons: { [key: string]: FontAwesomeIconProps } = {
```

Comment:
Can/should this be `[key: LogEntryIconKey]: ...`?

Reply:
Yeah it complaints about that but you can do ```[key in LogEntryIconKey]``` and it works